### PR TITLE
#1615 - Preferred masters integration

### PR DESCRIPTION
**Pull request by kvanhijf**

### CODE REVIEW

File: ovs/extensions/generic/watcher.py [hunk 1]
```diff
@@ -67,18 +67,16 @@ def services_running(self, target):
...
+                from ovs.extensions.db.arakoon.arakooninstaller import ArakoonInstaller, ArakoonClusterConfig
                 from ovs.extensions.db.arakoon.configuration import ArakoonConfiguration
-                from ovs.extensions.db.arakoon.ArakoonInstaller import ArakoonInstaller, ArakoonClusterConfig
                 from ovs.extensions.db.arakoon.pyrakoon.pyrakoon.compat import NoGuarantee
-                with open(ArakoonConfiguration.CACC_LOCATION) as config_file:
-                    contents = config_file.read()
-                config = ArakoonClusterConfig(cluster_id='cacc', load_config=False)
-                config.read_config(contents)
+
+                config = ArakoonClusterConfig(cluster_id='cacc', source_ip='127.0.0.1')
```
Comment (khenderick) [inline ovs/extensions/generic/watcher.py]: I think the `127.0.0.1` only works if you don't switch users. Might be the case here, but I don't like this "requirement".

File: ovs/lib/storagedriver.py [hunk 2]
```diff
@@ -225,7 +225,7 @@ def _on_demote(cluster_ip, master_ip, offline_node_ips=None):
         current_service = None
         remaining_ips = []
         for service in servicetype.services:
-            if service.name == 'arakoon-voldrv' and service.is_internal is True:  # Externally managed arakoon cluster service does not have storage router
+            if service.name == 'arakoon-voldrv' and service.is_internal is True:  # Externally managed arakoon cluster services do not have StorageRouter
```
Comment (khenderick) [inline ovs/lib/storagedriver.py]: `...s`

**Review by khenderick — CHANGES_REQUESTED** [review]

**Review by khenderick — APPROVED** [review]

**Comment by khenderick** [comment]
Last changes look good.

**Comment by kvanhijf** [comment]
https://github.com/openvstorage/framework/pull/1618 --> openvstorage-2.9.2-dev.1493800277.9147cba
https://github.com/openvstorage/alba-asdmanager/pull/248 --> openvstorage-sdm-1.9.0-dev.1493800370.72ae5e5
https://github.com/openvstorage/framework-alba-plugin/pull/405 --> openvstorage-backend-1.9.0-dev.1493800734.54d33bd

**Pull request merged**